### PR TITLE
BUG: to_datetime with infer_datetime_format dropped timezone names

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/tools/test_to_datetime.py [hunk 1]
```diff
@@ -1862,6 +1862,12 @@ def test_to_datetime_infer_datetime_format_series_start_with_nans(self, cache):
             pd.to_datetime(s, infer_datetime_format=True, cache=cache),
         )
 
+    def test_infer_datetime_format_tz_name(self):
+        s = pd.Series(["2019-02-02 08:07:13 UTC"])
```
Comment (jbrockmendel) [inline pandas/tests/tools/test_to_datetime.py]: is it just UTC that gets recognized?  "GMT"?  "UTC-3"?

**Review by jreback — APPROVED** [review]
lgtm. @jbrockmendel merge when good.

File: pandas/tests/tools/test_to_datetime.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/tools/test_to_datetime.py]: Recognizes UTC-3 but not GMT. I'll update with a parameterized test.

File: pandas/tests/tools/test_to_datetime.py [hunk 2]
```diff
@@ -1862,6 +1862,17 @@ def test_to_datetime_infer_datetime_format_series_start_with_nans(self, cache):
             pd.to_datetime(s, infer_datetime_format=True, cache=cache),
         )
 
+    @pytest.mark.parametrize(
+        "tz_name, offset", [("UTC", 0), ("UTC-3", 180), ("UTC+3", -180)]
+    )
+    def test_infer_datetime_format_tz_name(self, tz_name, offset):
```
Comment (jbrockmendel) [inline pandas/tests/tools/test_to_datetime.py]: GH ref?

**Comment by jbrockmendel** [comment]
LGTM pending GH ref

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**